server/resources/resource_management.go

```diff
@@ -518,9 +518,17 @@ func CreateResource(
 	}
 
 	//Validation
-	err = resourceSchema.ValidateOnCreate(dataMap)
-	if err != nil {
-		return ResourceError{err, fmt.Sprintf("Validation error: %s", err), WrongData}
+
+	if _, ok := context["go_validation"]; ok {
```

Comment:
@przemyslaw-dobrowolski-cl Could you answer this question?

Reply:
Validation of struct-based resources in golang must be different from map-based resources, as in JS. This comes from the fact, that in a structure in go you are not able to omit some fields so that they get the default value. In JS this can be done since default fields are simply omitted. I consider this change as not-a-perfect solution but it does the job.
One may think of having structures will all fields being optional but that makes the code messy and currently no better solution is known fot this problem.